qiskit_nature/second_q/mappers/bosonic_linear_mapper.py

```diff
@@ -0,0 +1,115 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Linear Mapper for Bosons."""
+
+from __future__ import annotations
+import operator
+
+from functools import reduce
+
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+from qiskit_nature.second_q.operators import BosonicOp
+from .bosonic_mapper import BosonicMapper
+
+
+class BosonicLinearMapper(BosonicMapper):
+    """
+    The Linear boson-to-qubit mapping.
+    This mapper implements the formula in Section II.C of: https://arxiv.org/abs/2108.04258
+
+    b_k^\\dagger -> sum(sqrt(n_k + 1) \\sigma_(n_k)^+\\sigma_(n_k + 1)^-, from n_k = 0 to n_k^max + 1)
+    where n_k^max is the truncation order (defined by the user)
+    In the following, we explicit the operators \\sigma^+ and \\sigma^- with the Pauli matrices:
+    \\sigma_(n_k)^+ := S_j^+ = 0.5 * (X_j + i*Y_j) ; \\sigma_(n_k)^- := S_j^- = 0.5 * (X_j - i*Y_j)
+
+    The length of the qubit register is: BosonicOp.num_mode * (BosonicLinearMapper.truncation + 1)
+
+    Generates a 'linear_encoding' of the Bosonic operator b_k^\\dagger, b_k to qubit operators
+        (linear combinations of pauli strings).
+        In this 'linear_encoding' each bosonic mode is represented via n_k^max + 1 qubits, where n_k^max
+        is the truncation of the mode (meaning the number of states used in the expansion of the mode,
+        or equivalently the state at which the maximum excitation can take place).
+        The mode |k> is then mapped to the occupation number vector
+        |0_{nk_^max}, 0_{nk_^max - 1}, ..., 0_{n_k + 1}, 1_{n_k}, 0_{n_k - 1}, ..., 0_{0_k}>
+    """
+
+    def _map_single(
+        self, second_q_op: BosonicOp, *, register_length: int | None = None
+    ) -> SparsePauliOp | PauliSumOp:
+        # If register_length was passed, override the one present in the BosonicOp
+        if register_length is None:
+            register_length = second_q_op.num_modes
+
+        qubit_register_length = register_length * (self.truncation + 1)
+        # Create a Pauli operator, which we will fill in this method
+        # TODO: Check if we need to normal/index_order
+        pauli_op: list[SparsePauliOp] = []
+        # Then we loop over all the terms of the bosonic operator
+        for terms, coeff in second_q_op.terms():
+            # Then loop over each term (terms -> List[Tuple[string, int]])
+            bos_op_to_pauli_op = SparsePauliOp(["I" * qubit_register_length], coeffs=[1.])
+            for op, idx in terms:
+                if op not in ('+', '-'):
+                    break
+                pauli_expansion: list[SparsePauliOp] = []
+                # Now we are dealing with a single bosonic operator. We have to perform the linear mapper
+                for n_k in range(self.truncation):
+                    prefactor = np.sqrt(n_k + 1) / 4
```

Comment:
```suggestion
                    prefactor = np.sqrt(n_k + 1) / 4.0
```
I like to be explicit about float division